src/components/menu-bar/menu-bar.jsx

```diff
@@ -152,6 +154,24 @@ class MenuBar extends React.Component {
             this.props.onRequestCloseEdit();
         };
     }
+    handleUpdateProject (updateFun) {
+        return () => {
+            this.props.onRequestCloseFile();
+            this.setState({projectSaveInProgress: true},
+                () => {
+                    updateFun().then(() => {
+                        this.setState({projectSaveInProgress: false});
+                    });
+                }
+            );
+        };
+    }
+    handleCloseFileMenuAndThen (fn) {
+        return () => {
+            this.props.onRequestCloseFile();
+            fn();
```

Comment:
This will also need error handling in the long run. Either as part of the fn(), or here if the fn returns an error.

Reply:
Probably so, which means that it probably shouldn't be in a generic handler like this. I'll give it it's own method.